### ISSUE TITLE
Add an HTML export

### ISSUE THREAD

**Issue by aloisdg** [tbvdm/sigtop#61]
Hello,

I am playing with the idea of an HTML export:

![image](https://github.com/user-attachments/assets/caccd7e2-a5f9-4387-bd07-c2a1cd9b6918)

the goal would be to have a gui export. It would be easier to read by non tech folks. What do you think?

**Comment by aloisdg** [comment]
My draft of solution can be seen and tried on my fork: https://github.com/aloisdg/sigtop/pull/1/files

If @tbvdm we can plan to add it sigtop later on.

Caveat: this is my first time writing Go without ever learning it.

**Comment by tbvdm** [comment]
Thanks for working on this! Very promising!

Please don't forget to escape HTML in the input; use [`html.EscapeString`](https://pkg.go.dev/html#EscapeString) for that. (It would be even better to use the [html/template](https://pkg.go.dev/html/template) package, but perhaps that's something for later.)

Also, I think HTML exports should be self-contained, so please avoid stylesheets from external sources.

You probably wanted to hack something together quickly, so I understand that you took a few shortcuts. :)

**Comment by aloisdg** [comment]
Yes this is indeed a quick a dirty version. I will remove watercss later on. I now next to nothing about Golang.

I would love to extract the html from the Go file and put in a proper html file. Feel free to help with that (with a package or file loading/import logic maybe?)

**Comment by tbvdm** [comment]
> I would love to extract the html from the Go file and put in a proper html file.

You can use the [embed](https://pkg.go.dev/embed) package:

```
import (
	_ "embed"
	// ...
)

//go:embed header.html
var htmlHeader string

func f() {
	fmt.Print(htmlHeader)
}
```

**Comment by tbvdm** [comment]
Great!

**Comment by idealemu** [comment]
It's been a while but, any updates on this ? Or help needed ? :)

**Comment by ldexterldesign** [comment]
👋 all,

Hope y'all OK!

@tbvdm thanks for great software 💌

[imessage-exporter] - which is awesome, @ReagentX also did a great job, has this feature - perhaps it could be integrated here as I'm sure many people are (like me) are aiming to move away from Apple to Signal for a privacy upgrade..?

Hope this helps

🤞

[imessage-exporter]: https://github.com/ReagentX/imessage-exporter
